proxy/dns_context.go

```diff
@@ -61,6 +61,8 @@ type DNSContext struct {
 	// udpSize is the UDP buffer size from request's EDNS0 RR if presented,
 	// or default otherwise.
 	udpSize uint16
+	// Temporarily disable cache reads for prefetching
+	disableCacheReading bool
```

Comment:
Instead of adding this flag, I'd prefer to refactor the `Resolve` method, split the resolution and the cache-specific logic.

Smth like this:

```
func Resolve() {
    ... check cache here ...

    // communite with upstreams
    p.exchange()

    ... save to cache if needed ... 
}
```